Dockerfile

```diff
@@ -2,8 +2,8 @@
 ARG base_image=public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424
 # Build the manager binary
 # TODO(vijtrip2) move this builder image to public.ecr.aws/eks-distro-build-tooling/builder-base, when builder-base
-# supports golang 1.17
-FROM public.ecr.aws/bitnami/golang:1.17 as builder
+# supports golang 1.19
+FROM public.ecr.aws/docker/library/golang:1.19.0 as builder
```

Comment:
I'm very surprised that this image change fixes it. Were you trying to build the image in an arm machine?

Reply:
@acornett21 pointed out that he observed a problem when trying to build a controller using Go `v1.19`

Reply:
I added the bug as a relates as a comment below. 

But as it relates to this PR, maybe I don't understand the problem. But just updating this value would not support multiarch. The binaries for the controller would also need to be built to support multi-arch (they might already be and I'm not aware of that). Also, just making the container go 1.19 and not updating all of the `mod` files in the the `code-gen` ,`runtime`, and all the controllers seems like the wrong approach. I'd think we need to be `all or nothing` on this type of update. Those are just some thoughts I have.

Reply:
Thats where I build the image
```
[ec2-user@ip-192-168-38-57 ~]$ uname -a
Linux ip-192-168-38-57.us-west-2.compute.internal 5.10.130-118.517.amzn2.aarch64 #1 SMP Wed Jul 13 16:51:59 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux```

Reply:
Output of build
```
[root@ip-192-168-38-57 code-generator]# ./scripts/build-controller-image.sh rds
building 'rds' controller docker image with tag: aws-controllers-k8s:rds-v0.0.30-12-g81733b3
 git commit: 81733b31e99a59a900c6ee071608960715bb241f
[+] Building 179.3s (20/20) FINISHED                                                                                                                                                
 => [internal] load build definition from Dockerfile                                                                                                                           0.0s
 => => transferring dockerfile: 2.37kB                                                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                              0.0s
 => => transferring context: 2B                                                                                                                                                0.0s
 => [internal] load metadata for public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424                                                 0.1s
 => [internal] load metadata for public.ecr.aws/docker/library/golang:latest                                                                                                   0.7s
 => [auth] aws:: docker/library/golang:pull token for public.ecr.aws                                                                                                           0.0s
 => [builder  1/11] FROM public.ecr.aws/docker/library/golang:latest@sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4e70607ffa05f72466d6                                  19.9s
 => => resolve public.ecr.aws/docker/library/golang:latest@sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4e70607ffa05f72466d6                                             0.0s
 => => sha256:5fd2567d7c49ac7447b1be01873c5b50d363463689e2395efeb5795eac9008f2 1.80kB / 1.80kB                                                                                 0.0s
 => => sha256:114ba63dd73a866ac1bb59fe594dfd218f44ac9b4fa4b2c68499da5584fcfa9d 53.68MB / 53.68MB                                                                               4.4s
 => => sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4e70607ffa05f72466d6 2.35kB / 2.35kB                                                                                 0.0s
 => => sha256:bc0b8a8acead4d7bf71c232054b2a0a8e08eb3e1e2cf46f9f3dba68723e88c85 5.15MB / 5.15MB                                                                                 1.0s
 => => sha256:a4ea641ee67989acdb6af3d8b9b984ecd751a2a83c3b7ce071542b31c9ac1304 10.66MB / 10.66MB                                                                               0.9s
 => => sha256:26b88b86d196c85f18b7a014d9d6a0e449e5d60dbd5a46af56289e711bbc8d37 7.10kB / 7.10kB                                                                                 0.0s
 => => sha256:04e9e95aca686baf2f9b88f3822104381dc1cd0e2bd6dc105b7468059336dbab 54.68MB / 54.68MB                                                                               5.5s
 => => sha256:08bf97f348f4f995ce74ff00207e68e1198e1b694048c6b6572b3752f84ea020 81.12MB / 81.12MB                                                                               6.8s
 => => extracting sha256:114ba63dd73a866ac1bb59fe594dfd218f44ac9b4fa4b2c68499da5584fcfa9d                                                                                      2.8s
 => => sha256:32ea6dde57d67835cf006d941f66545f2ceba7be0b2630fc6d9db893bfffce92 115.07MB / 115.07MB                                                                            12.1s
 => => sha256:8f5090d68e6e5d3ad8814f2f5285758a2c4b2bf05598450a327154519655a0ee 125B / 125B                                                                                     5.7s
 => => extracting sha256:bc0b8a8acead4d7bf71c232054b2a0a8e08eb3e1e2cf46f9f3dba68723e88c85                                                                                      0.3s
 => => extracting sha256:a4ea641ee67989acdb6af3d8b9b984ecd751a2a83c3b7ce071542b31c9ac1304                                                                                      0.3s
 => => extracting sha256:04e9e95aca686baf2f9b88f3822104381dc1cd0e2bd6dc105b7468059336dbab                                                                                      2.8s
 => => extracting sha256:08bf97f348f4f995ce74ff00207e68e1198e1b694048c6b6572b3752f84ea020                                                                                      2.9s
 => => extracting sha256:32ea6dde57d67835cf006d941f66545f2ceba7be0b2630fc6d9db893bfffce92                                                                                      5.5s
 => => extracting sha256:8f5090d68e6e5d3ad8814f2f5285758a2c4b2bf05598450a327154519655a0ee                                                                                      0.0s
 => CACHED [stage-1 1/3] FROM public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424@sha256:defed268f03173500174d285a8bbb5dd98d31b5f8f  0.0s
 => [internal] load build context                                                                                                                                              0.0s
 => => transferring context: 7.28kB                                                                                                                                            0.0s
 => [builder  2/11] WORKDIR /github.com/aws-controllers-k8s/rds-controller                                                                                                     0.4s
 => [builder  3/11] COPY rds-controller/LICENSE /github.com/aws-controllers-k8s/rds-controller/LICENSE                                                                         0.0s
 => [builder  4/11] COPY rds-controller/ATTRIBUTION.md /github.com/aws-controllers-k8s/rds-controller/ATTRIBUTION.md                                                           0.0s
 => [builder  5/11] COPY rds-controller/go.mod /github.com/aws-controllers-k8s/rds-controller/go.mod                                                                           0.0s
 => [builder  6/11] COPY rds-controller/go.sum /github.com/aws-controllers-k8s/rds-controller/go.sum                                                                           0.0s
 => [builder  7/11] RUN go mod download                                                                                                                                       15.5s
 => [builder  8/11] COPY rds-controller/apis /github.com/aws-controllers-k8s/rds-controller/apis                                                                               0.0s
 => [builder  9/11] COPY rds-controller/cmd /github.com/aws-controllers-k8s/rds-controller/cmd                                                                                 0.0s
 => [builder 10/11] COPY rds-controller/pkg /github.com/aws-controllers-k8s/rds-controller/pkg                                                                                 0.2s
 => [builder 11/11] RUN GIT_VERSION=v0.0.30-12-g81733b3 &&     GIT_COMMIT=81733b31e99a59a900c6ee071608960715bb241f &&     BUILD_DATE=2022-08-17T22:47 &&     go build -ldfl  142.0s
 => [stage-1 2/3] COPY --from=builder /github.com/aws-controllers-k8s/rds-controller/bin/controller /github.com/aws-controllers-k8s/rds-controller/LICENSE /github.com/aws-co  0.1s
 => exporting to image                                                                                                                                                         0.1s
 => => exporting layers                                                                                                                                                        0.1s
 => => writing image sha256:cd1ad0a2070ac2360775ab2baa5903666704f970493cb53827c5d1a51aaa1e7d                                                                                   0.0s
 => => naming to docker.io/library/aws-controllers-k8s:rds-v0.0.30-12-g81733b3                                                                                                 0.0s
```

Reply:
Also the docker run output
```
[root@ip-192-168-38-57 code-generator]# docker run -it docker.io/library/aws-controllers-k8s:rds-v0.0.30-12-g81733b3 /bin/bash
{"level":"error","ts":"2022-08-18T18:43:03.455Z","logger":"setup","msg":"Unable to create controller manager","aws.service":"rds","error":"unable to start service controller as AWS region is missing. Please pass --aws-region flag or set AWS_REGION environment variable","stacktrace":"main.main\n\t/github.com/aws-controllers-k8s/rds-controller/cmd/controller/main.go:74\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
```

Reply:
@A-Hilaly you are right. I forgot to add the GOARCH modification. I hard coded it and skipped it as GOARCH uses `amd64` for `x86_64` and `arm64` for `aarch64`. Now included with modified https://github.com/aws-controllers-k8s/code-generator/blob/main/scripts/build-controller-image.sh. I tested it with EC2 instances with `64-bit(x86)` and `64-bit(Arm)`

Reply:
Thats the test with x86_64
```
[root@ip-192-168-44-195 code-generator]# service docker start
Redirecting to /bin/systemctl start docker.service
[root@ip-192-168-44-195 code-generator]# ./scripts/build-controller-image.sh rds
building 'rds' controller docker image with tag: aws-controllers-k8s:rds-v0.0.30-12-g81733b3
 git commit: 81733b31e99a59a900c6ee071608960715bb241f
[+] Building 69.0s (18/20)                                                                                                       
 => => extracting sha256:9daef329d35093868ef75ac8b7c6eb407fa53abbcb3a264c218c2ec7bca716e6                                   2.4s
 => => extracting sha256:1c28274a8e7c4c48dd6843a6c33a0192271cfc7ef94f059ef7d70c4b60da6702                                   2.7s
 => => extracting sha256:39a7c004eb589546d4c09ebb919be09a745943dd12073b6bb100bd029e0a2f90                                   6.0s
 => => extracting sha256:a98825e7f2fd4a73a05ae2361e3d68cc7919e6b4e47e8ccd97554d8d01dfd840                                   0.0s
 => [stage-1 1/3] FROM public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424@sha25  3.4s
 => => resolve public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424@sha256:defed2  0.0s
 => => sha256:defed268f03173500174d285a8bbb5dd98d31b5f8f894736e936740c2953baff 703B / 703B                                  0.0s
 => => sha256:524e12ba594336d3f759726a1235400d966f5f781e78f0a378b9d8abf68b6744 505B / 505B                                  0.0s
[+] Building 143.9s (21/21) FINISHED                                                                                             
 => [internal] load build definition from Dockerfile                                                                        0.0s
 => => transferring dockerfile: 2.32kB                                                                                      0.0s
 => [internal] load .dockerignore                                                                                           0.0s
 => => transferring context: 2B                                                                                             0.0s
 => [internal] load metadata for public.ecr.aws/docker/library/golang:1.19.0                                                0.5s
 => [internal] load metadata for public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-163832  0.5s
 => [auth] aws:: docker/library/golang:pull token for public.ecr.aws                                                        0.0s
 => [auth] aws:: eks-distro-build-tooling/eks-distro-minimal-base-nonroot:pull token for public.ecr.aws                     0.0s
 => [builder  1/11] FROM public.ecr.aws/docker/library/golang:1.19.0@sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4  17.1s
 => => resolve public.ecr.aws/docker/library/golang:1.19.0@sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4e70607ffa05  0.0s
 => => sha256:684d791f8fd1b28485ee32239d92cde7099d97c88c5ed2e955f65f4adcf35d0f 1.80kB / 1.80kB                              0.0s
 => => sha256:da8d5a6f7a03ed3c735d22d0c098762a29b470df324315d5248ac8f6b3177235 7.08kB / 7.08kB                              0.0s
 => => sha256:d9d4b9b6e964657da49910b495173d6c4f0d9bc47b3b44273cf82fd32723d165 5.16MB / 5.16MB                              0.3s
 => => sha256:1860373709cc04bacfcf3bb1aaa8c14bb1243e971e9c4e70607ffa05f72466d6 2.35kB / 2.35kB                              0.0s
 => => sha256:001c52e26ad57e3b25b439ee0052f6692e5c0f2d5d982a00a8819ace5e521452 55.00MB / 55.00MB                            1.5s
 => => sha256:2068746827ec1b043b571e4788693eab7e9b2a95301176512791f8c317a2816a 10.88MB / 10.88MB                            0.5s
 => => sha256:9daef329d35093868ef75ac8b7c6eb407fa53abbcb3a264c218c2ec7bca716e6 54.58MB / 54.58MB                            2.3s
 => => sha256:1c28274a8e7c4c48dd6843a6c33a0192271cfc7ef94f059ef7d70c4b60da6702 85.90MB / 85.90MB                            4.4s
 => => sha256:39a7c004eb589546d4c09ebb919be09a745943dd12073b6bb100bd029e0a2f90 148.80MB / 148.80MB                          9.8s
 => => extracting sha256:001c52e26ad57e3b25b439ee0052f6692e5c0f2d5d982a00a8819ace5e521452                                   3.0s
 => => sha256:a98825e7f2fd4a73a05ae2361e3d68cc7919e6b4e47e8ccd97554d8d01dfd840 156B / 156B                                  2.4s
 => => extracting sha256:d9d4b9b6e964657da49910b495173d6c4f0d9bc47b3b44273cf82fd32723d165                                   0.2s
 => => extracting sha256:2068746827ec1b043b571e4788693eab7e9b2a95301176512791f8c317a2816a                                   0.3s
 => => extracting sha256:9daef329d35093868ef75ac8b7c6eb407fa53abbcb3a264c218c2ec7bca716e6                                   2.4s
 => => extracting sha256:1c28274a8e7c4c48dd6843a6c33a0192271cfc7ef94f059ef7d70c4b60da6702                                   2.7s
 => => extracting sha256:39a7c004eb589546d4c09ebb919be09a745943dd12073b6bb100bd029e0a2f90                                   6.0s
 => => extracting sha256:a98825e7f2fd4a73a05ae2361e3d68cc7919e6b4e47e8ccd97554d8d01dfd840                                   0.0s
 => [stage-1 1/3] FROM public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424@sha25  3.4s
 => => resolve public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2021-12-01-1638322424@sha256:defed2  0.0s
 => => sha256:defed268f03173500174d285a8bbb5dd98d31b5f8f894736e936740c2953baff 703B / 703B                                  0.0s
 => => sha256:524e12ba594336d3f759726a1235400d966f5f781e78f0a378b9d8abf68b6744 505B / 505B                                  0.0s
 => => sha256:6f9a39e74d14442d5964d51864af0fa06485db9e39a08f408db7d04c8120e820 892B / 892B                                  0.0s
 => => sha256:03e1f5a9556d6cefb3b1efb1e5b5b19a347e7057a18ec1f5baddd3eaa7c243e6 1.86MB / 1.86MB                              2.5s
 => => extracting sha256:03e1f5a9556d6cefb3b1efb1e5b5b19a347e7057a18ec1f5baddd3eaa7c243e6                                   0.8s
 => [internal] load build context                                                                                           0.1s
 => => transferring context: 1.27MB                                                                                         0.0s
 => [builder  2/11] WORKDIR /github.com/aws-controllers-k8s/rds-controller                                                  0.5s
 => [builder  3/11] COPY rds-controller/LICENSE /github.com/aws-controllers-k8s/rds-controller/LICENSE                      0.0s
 => [builder  4/11] COPY rds-controller/ATTRIBUTION.md /github.com/aws-controllers-k8s/rds-controller/ATTRIBUTION.md        0.0s
 => [builder  5/11] COPY rds-controller/go.mod /github.com/aws-controllers-k8s/rds-controller/go.mod                        0.0s
 => [builder  6/11] COPY rds-controller/go.sum /github.com/aws-controllers-k8s/rds-controller/go.sum                        0.0s
 => [builder  7/11] RUN go mod download                                                                                    10.8s
 => [builder  8/11] COPY rds-controller/apis /github.com/aws-controllers-k8s/rds-controller/apis                            0.0s
 => [builder  9/11] COPY rds-controller/cmd /github.com/aws-controllers-k8s/rds-controller/cmd                              0.0s
 => [builder 10/11] COPY rds-controller/pkg /github.com/aws-controllers-k8s/rds-controller/pkg                              0.0s
 => [builder 11/11] RUN GIT_VERSION=v0.0.30-12-g81733b3 &&     GIT_COMMIT=81733b31e99a59a900c6ee071608960715bb241f &&     113.9s
 => [stage-1 2/3] COPY --from=builder /github.com/aws-controllers-k8s/rds-controller/bin/controller /github.com/aws-contro  0.1s
 => exporting to image                                                                                                      0.3s
 => => exporting layers                                                                                                     0.3s
 => => writing image sha256:62507607c1ec3f831a3ee8c36228861d1a26cf2faad1be251bafa93b076aec65                                0.0s
 => => naming to docker.io/library/aws-controllers-k8s:rds-v0.0.30-12-g81733b3                                              0.0s
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# 
[root@ip-192-168-44-195 code-generator]# docker run -it docker.io/library/aws-controllers-k8s:rds-v0.0.30-12-g81733b3 /bin/bash
{"level":"error","ts":"2022-08-18T19:56:50.405Z","logger":"setup","msg":"Unable to create controller manager","aws.service":"rds","error":"unable to start service controller as AWS region is missing. Please pass --aws-region flag or set AWS_REGION environment variable","stacktrace":"main.main\n\t/github.com/aws-controllers-k8s/rds-controller/cmd/controller/main.go:74\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
[root@ip-192-168-44-195 code-generator]# uname -i
x86_64
[root@ip-192-168-44-195 code-generator]# 
```